lib/base_jobmaster.go

```diff
@@ -203,6 +209,17 @@ func (d *DefaultBaseJobMaster) IsMasterReady() bool {
 	return d.master.IsMasterReady()
 }
 
+func (d *DefaultBaseJobMaster) Exit(ctx context.Context, status WorkerStatus, err error) error {
```

Comment:
how to make sure this method is called by the jobmaster, if not ,the jobmanager can't recognized which jobmaster should enter the failover process

Reply:
In current design it must be called from job master (user logic)
- If user logic doesn't call `Exit`, the job *can't finish*
- Besides, if job master meets error and exits, the job manager can failover it as before.

Reply:
it‘s not reasonable now ,since as designed ,the executor will cal the tick method of the worker , if the tick method return error , the executor will assume the worker in the finished status or error status. then store the worker's status and remove it from the runtime

Reply:
executor doesn't access the kv meta of worker status

Reply:
The `Exit` is designed to be called in `Tick` of worker / job master